Tests/AsyncChannelsTests/AsyncChannelTests.swift

```diff
@@ -266,6 +266,112 @@ final class AsyncTest: XCTestCase {
         let r = await result.reduce(into: []) { $0.append($1) }
         XCTAssertEqual(["foo", "bar"].sorted(), r.sorted())
     }
+    
+    func testOptionalSomeSelect() async {
+        let a = Channel<String>()
+        let result = Channel<String>(capacity: 1)
+        
+        Task {
+            await a <- "foo"
+        }
+        
+        await select {
+            if true {
+                receive(a) { await result <- $0! }
+            }
+        }
+
+        result.close()
+        
+        let r = await result.reduce(into: []) { $0.append($1) }
```

Comment:
For single result asserts, you can use the helper: 
```swift
await assertChanRx(result, "foo")
```
This avoids the need for the reduce statement. 
(Same for other cases below)

Reply:
I fixed it at [80a72f6](https://github.com/gh123man/Async-Channels/pull/10/commits/80a72f671a200fc3b02754d1e9d05049c844c8bc).